skyportal/handlers/thumbnail.py

```diff
@@ -0,0 +1,129 @@
+          - in: path
+            name: data
+            schema:
+              type: string
+              format: byte
+            description: base64-encoded PNG image file contents. Image size must be between 100px and 500px on a side.
+          - in: path
+            name: ttype
+            schema:
+              type: string
+            description: Thumbnail type. Must be one of 'new', 'ref', 'sub', 'sdss', 'dr8', 'new_gz', 'ref_gz', 'sub_gz'
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - Success
+                    - type: object
+                      properties:
+                        id:
+                          type: int
+                          description: New thumbnail ID
+        """
+        data = self.get_json()
+        if 'photometry_id' in data:
+            phot = Photometry.query.get(int(data['photometry_id']))
+            source = phot.source
+        elif 'source_id' in data:
+            source = Source.query.get(data['source_id'])
+            try:
+                phot = source.photometry[0]
+            except IndexError:
+                return self.error('Specified source does not yet have any photometry data.')
+        else:
+            return self.error('One of either source_id or photometry_id are required.')
+        basedir = Path(os.path.dirname(__file__))/'..'/'..'
+        if os.path.abspath(basedir).endswith('skyportal/skyportal'):
+            basedir = basedir/'..'
+        file_uri = os.path.abspath(
+            basedir/f'static/thumbnails/{source.id}_{data["ttype"]}.png')
+        if not os.path.exists(os.path.dirname(file_uri)):
+            (basedir/'static/thumbnails').mkdir(parents=True)
+        file_bytes = base64.b64decode(data['data'])
+        im = Image.open(io.BytesIO(file_bytes))
+        if im.format != 'PNG':
+            return self.error('Invalid thumbnail image type. Only PNG are supported.')
+        if not (100, 100) <= im.size <= (500, 500):
+            return self.error('Invalid thumbnail size. Only thumbnails '
+                              'between (100, 100) and (500, 500) allowed.')
+        try:
+            t = Thumbnail(type=data['ttype'],
+                          photometry=phot,
+                          file_uri=file_uri,
+                          public_url=f'/static/thumbnails/{source.id}_{data["ttype"]}.png')
+            DBSession.add(t)
+        except TypeError:
+            return self.error('Invalid thumbnail type. Please refer to '
+                              'API docs for a list of allowed types.')
+        except Exception as e:
+            return self.error(f'Error creating thumbnail: {str(e)}. Please check '
+                              'submitted values against API docs.')
+        with open(file_uri, 'wb') as f:
+            f.write(file_bytes)
```

Comment:
@stefanv  do you mean factoring this part out so it can be called in both handlers?

Reply:
Yes, I think ideally we do not want duplicated logic like this floating around.

Reply:
But there may be a way to call the one post method from another.